plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/popup/QInlineCompletionProvider.kt

```diff
@@ -391,8 +401,22 @@ class QInlineCompletionProvider(private val cs: CoroutineScope) : InlineCompleti
 
     override suspend fun getSuggestion(request: InlineCompletionRequest): InlineCompletionSuggestion {
         val editor = request.editor
-        val document = editor.document
         val project = editor.project ?: return InlineCompletionSuggestion.Empty
+
+        // try to refresh automatically if possible, otherwise ask user to login again
+        if (isQExpired(project)) {
```

Comment:
this `isQExpired` should be very fast so that inline completion latency does not go up

Reply:
yes this is just check on token timestamp